src/hotspot/os/linux/osContainer_linux.cpp

```diff
@@ -68,6 +70,29 @@ void OSContainer::init() {
 
   _is_containerized = true;
 
+  if (getpid() == 1) {
+    // This process is in container
+    _runtime = os::strdup_check_oom(getenv("container"));
```

Comment:
In my testing this shows `oci`:

```
$ podman run --rm -ti fedora:33
[root@2322a30ef7cd /]# echo $container
oci
```

So I'm not sure this will be very helpful. Systemd does some fairly involved translation:
https://github.com/systemd/systemd/blob/68337e55f62cf49b7bdfb73dc5662e23b0ea17fa/src/basic/virt.c#L677

Those heuristics will involve a partial implementation of https://bugs.openjdk.java.net/browse/JDK-8261242

Also consider that there are multiple container runtimes when podman is in use (I don't know about docker). For example `crun` and `runc`. In a way, container runtime then becomes ambiguous too.

Reply:
I think it is ok if we can know the process was run on a container. It is better to know what runtime was used if possible - it might not be mandatory. So I think we can refer /run/.containerenv and /.dockerenv now like systemd. It might not be enough, but it would be acceptable.

JDK-8261242 has been assigned to Harold, so I will not start to work for it now.
(I can work for it of course if you agree with the above.)